### PR TITLE
Convert Attribute editor to JavaFx pane and use TopComponent framework

### PR DESCRIPTION
**Pull request by aldebaran30701**
### Description of the Change
Add final modifiers
convert swing panel to pane
switch over to top component framework for graph listeners

### Alternate Designs
N/A

### Why Should This Be In Core?
Code consistency

### Benefits
Code consistency

### Possible Drawbacks
None realized

### Verification Process
Open/close - state testing between graphs.

### Applicable Issues
#332

### CODE REVIEW

**Comment by aldebaran30701** [comment]
> remove unused imports automatically on save

This will be a huge time-saver! Much appreciated

File: CoreAttributeEditorView/src/au/gov/asd/tac/constellation/views/attributeeditor/AttributeEditorTopComponent.java [hunk 1]
```diff
@@ -248,7 +248,7 @@ public UndoRedo getUndoRedo() {
     }
 
     @Override
-    public void preferenceChange(PreferenceChangeEvent event) {
+    protected void handlePreferenceChange(final PreferenceChangeEvent event) {
```
Comment (cygnus-x-1) [inline CoreAttributeEditorView/src/au/gov/asd/tac/constellation/views/attributeeditor/AttributeEditorTopComponent.java:251]: Again, this should be handled by the framework.

File: CoreAttributeEditorView/src/au/gov/asd/tac/constellation/views/attributeeditor/AttributeEditorPanel.java [hunk 2]
```diff
@@ -212,21 +207,13 @@ public AttributeEditorPanel(AttributeEditorTopComponent parent) {
...
-
         Platform.setImplicitExit(false);
         Platform.runLater(() -> {
             root = new StackPane();
-
-            final Scene scene = new Scene(root, 0, height);
-            scene.getStylesheets().add(JavafxStyleManager.getMainStyleSheet());
-            scene.rootProperty().get().setStyle(String.format("-fx-font-size:%d;", FontUtilities.getOutputFontSize()));
-            scene.getStylesheets().add(AttributeEditorPanel.class.getResource(DARK_THEME).toExternalForm());
+            root.getStylesheets().add(JavafxStyleManager.getMainStyleSheet());
```
Comment (cygnus-x-1) [inline CoreAttributeEditorView/src/au/gov/asd/tac/constellation/views/attributeeditor/AttributeEditorPanel.java:215]: This should be handled by the framework - it'll automatically apply the default stylesheet, and additionally apply a stylesheet specified by `JavaFxTopComponent.createStyle`. These styles will cascade down from the top component to its children (at least that's how it's supposed to work.

File: CoreAttributeEditorView/src/au/gov/asd/tac/constellation/views/attributeeditor/AttributeEditorTopComponent.java (hunk 1 above)
Comment (aldebaran30701) [reply CoreAttributeEditorView/src/au/gov/asd/tac/constellation/views/attributeeditor/AttributeEditorTopComponent.java:251]: Neat!

File: CoreAttributeEditorView/src/au/gov/asd/tac/constellation/views/attributeeditor/AttributeEditorPanel.java (hunk 2 above)
Comment (aldebaran30701) [reply CoreAttributeEditorView/src/au/gov/asd/tac/constellation/views/attributeeditor/AttributeEditorPanel.java:215]: It seems to be working fine after putting on the correct style sheet either way. Next commit will have the fixes in it 👍

**Pull request merged**